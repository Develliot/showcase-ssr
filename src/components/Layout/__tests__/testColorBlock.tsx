import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ColorBlock } from '../';

it('renders correctly with children', () => {
    const wrapper: ShallowWrapper = shallow(
        <ColorBlock>
            <div>Hello</div>
            <div>World</div>
        </ColorBlock>
    );
    expect(wrapper).toMatchSnapshot();
});

it('renders correctly a set color', () => {
    const wrapper: ShallowWrapper = shallow(
        <ColorBlock color='pink'>
            <div>Hello</div>
            <div>World</div>
        </ColorBlock>
    );
    expect(wrapper).toMatchSnapshot();
});

it('renders without padding', () => {
    const wrapper: ShallowWrapper = shallow(
        <ColorBlock padding={false}>
            <div>Hello</div>
            <div>World</div>
        </ColorBlock>
    );
    expect(wrapper).toMatchSnapshot();
});
