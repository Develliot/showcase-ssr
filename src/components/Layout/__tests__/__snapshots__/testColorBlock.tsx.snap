// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly a set color 1`] = `
<Layout__Block
  color="pink"
>
  <Layout__VerticalSpacer
    size="extraLarge"
  />
  <div>
    Hello
  </div>
  <div>
    World
  </div>
  <Layout__VerticalSpacer
    size="extraExtraLarge"
  />
</Layout__Block>
`;

exports[`renders correctly with children 1`] = `
<Layout__Block
  color="white"
>
  <Layout__VerticalSpacer
    size="extraLarge"
  />
  <div>
    Hello
  </div>
  <div>
    World
  </div>
  <Layout__VerticalSpacer
    size="extraExtraLarge"
  />
</Layout__Block>
`;

exports[`renders without padding 1`] = `
<Layout__Block
  color="white"
>
  <div>
    Hello
  </div>
  <div>
    World
  </div>
</Layout__Block>
`;
